www/js/lib/archive.js

```diff
@@ -388,21 +368,26 @@ define(['normalize_string', 'geometry', 'title', 'util', 'titleIterators', 'jque
             var iterator = new titleIterators.SequentialTitleIterator(that, offset);
             function addNext() {
                 if (titles.length >= maxSize) {
-                    return jQuery.Deferred().resolve(titles, maxSize);
+                    callbackFunction(titles, maxSize);
+                    return 1;
                 }
                 return iterator.advance().then(function(title) {
-                    if (title === null)
-                        return jQuery.Deferred().resolve(titles, maxSize);
+                    if (title === null) {
+                        callbackFunction(titles, maxSize);
+                        return 1;
+                    }
                     // check whether this title really starts with the prefix
                     var name = normalize(title._name);
-                    if (name.length < prefix.length || name.substring(0, prefix.length) !== prefix)
-                        return jQuery.Deferred().resolve(titles, maxSize);
+                    if (name.length < prefix.length || name.substring(0, prefix.length) !== prefix) {
+                        callbackFunction(titles, maxSize);
+                        return 1;
+                    }
                     titles.push(title);
                     return addNext();
                 });
             }
             return addNext();
-        }).then(callbackFunction, errorHandler);
+        }).then(function(){}, errorHandler);
```

Comment:
Is it on purpose that you replaced the call to callbackFunction by an empty function?